reactive/kotlinx-coroutines-reactive/src/flow/PublisherAsFlow.kt

```diff
@@ -70,7 +71,7 @@ private class PublisherAsFlow<T : Any>(
 
     override suspend fun collect(collector: FlowCollector<T>) {
         val subscriber = ReactiveSubscriber<T>(capacity, requestSize)
-        publisher.subscribe(subscriber)
+        publisher.injectCoroutineContext(coroutineContext).subscribe(subscriber)
```

Comment:
This injection is missing from the `produceImpl`.
Please add a failing test first

Reply:
Test added, injection fixed.